core/unit_test/TestViewCopy_c.hpp

```diff
@@ -0,0 +1,377 @@
+}
+
+template <typename ExecSpace, typename DT, typename... DP>
+bool check_magic_value(
+    const ExecSpace& space, const Kokkos::View<DT, DP...>& a, int magic,
+    std::enable_if_t<Kokkos::ViewTraits<DT, DP...>::rank == 1>* = nullptr) {
+  auto policy = Kokkos::RangePolicy<ExecSpace>(space, 0, a.extent(0));
+
+  bool all_elements_are_set;  // Unititialized, set by parallel_reduce
+  Kokkos::fence();
+  Kokkos::parallel_reduce(
+      "TestViewCopy::CheckMagicValue", policy,
+      KOKKOS_LAMBDA(int i, bool& lcl_is) {
+        lcl_is = lcl_is && (a(i) == magic);
+      },
+      Kokkos::LAnd<bool>(all_elements_are_set));
+  space.fence();
+  return all_elements_are_set;
+}
+
+template <typename ExecSpace, typename DT, typename... DP>
+bool check_magic_value(
+    const ExecSpace& space, const Kokkos::View<DT, DP...>& a, int magic,
+    std::enable_if_t<Kokkos::ViewTraits<DT, DP...>::rank == 2>* = nullptr) {
+  auto policy = Kokkos::MDRangePolicy<Kokkos::Rank<2>, ExecSpace>(
+      space, {0, 0}, {a.extent(0), a.extent(1)});
+
+  bool all_elements_are_set;  // Unititialized, set by parallel_reduce
+  Kokkos::fence();
+  Kokkos::parallel_reduce(
+      "TestViewCopy::CheckMagicValueRank2", policy,
+      KOKKOS_LAMBDA(int i0, int i1, bool& lcl_is) {
+        lcl_is = lcl_is && (a(i0, i1) == magic);
+      },
+      Kokkos::LAnd<bool>(all_elements_are_set));
+  space.fence();
+  return all_elements_are_set;
+}
+
+template <typename ExecSpace, typename DT, typename... DP>
+bool check_magic_value(
+    const ExecSpace& space, const Kokkos::View<DT, DP...>& a, int magic,
+    std::enable_if_t<Kokkos::ViewTraits<DT, DP...>::rank == 3>* = nullptr) {
+  auto policy = Kokkos::MDRangePolicy<Kokkos::Rank<3>, ExecSpace>(
+      space, {0, 0, 0}, {a.extent(0), a.extent(1), a.extent(2)});
+
+  bool all_elements_are_set;  // Unititialized, set by parallel_reduce
+  Kokkos::fence();
+  Kokkos::parallel_reduce(
+      "TestViewCopy::CheckMagicValueRank3", policy,
+      KOKKOS_LAMBDA(int i0, int i1, int i2, bool& lcl_is) {
+        lcl_is = lcl_is && (a(i0, i1, i2) == magic);
+      },
+      Kokkos::LAnd<bool>(all_elements_are_set));
+  space.fence();
+  return all_elements_are_set;
+}
+
+template <typename ExecSpace, typename DT, typename... DP>
+bool check_magic_value(
+    const ExecSpace& space, const Kokkos::View<DT, DP...>& a, int magic,
+    std::enable_if_t<Kokkos::ViewTraits<DT, DP...>::rank == 4>* = nullptr) {
+  auto policy = Kokkos::MDRangePolicy<Kokkos::Rank<4>, ExecSpace>(
+      space, {0, 0, 0, 0},
+      {a.extent(0), a.extent(1), a.extent(2), a.extent(3)});
+
+  bool all_elements_are_set;  // Unititialized, set by parallel_reduce
+  Kokkos::fence();
+  Kokkos::parallel_reduce(
+      "TestViewCopy::CheckMagicValueRank4", policy,
+      KOKKOS_LAMBDA(int i0, int i1, int i2, int i3, bool& lcl_is) {
+        lcl_is = lcl_is && (a(i0, i1, i2, i3) == magic);
+      },
+      Kokkos::LAnd<bool>(all_elements_are_set));
+  space.fence();
```

Comment:
I don't think this fence is necessary.

Reply:
For this one, it is not very clear for me.

The [documentation](https://kokkos.org/kokkos-core-wiki/API/core/parallel-dispatch/parallel_reduce.html#semantics) states
> The call is potentially asynchronous if the ReducerArgument is not a scalar type.

While here it is not a view, I thought it qualified as a `Reducer` and so was not a scalar, so the reduction was potentially asynchronous?

Reply:
The reducer references a scalar, i.e., `reducer.references_scalar()` is true, and in this case, we still fence inside `Kokkos`.

Reply:
Ok, thanks for the explanation.